lib/docker-sync/sync_manager.rb

```diff
@@ -121,10 +123,16 @@ def run(sync_name = nil)
 
     def join_threads
       begin
+        has_threads = false
         @sync_processes.each do |sync_process|
-          sync_process.watch_thread.join unless sync_process.watch_thread.nil?
+          if sync_process.watch_thread
+            sync_process.watch_thread.join
+            has_threads = true
+          end
         end
 
+        sleep unless has_threads
```

Comment:
As I said previously in the first message of the PR.

> I wasn't able to get unison working as a new thread, I needed to fork it. This is a problem because, the .join method no longer prevents the main process to exit if you use only this strategy. I use a silly hack with sleep, sure there is better implementation than that.

If you don't sleep, the main process exits and you have no way to kill started unison process automatically.